File: src/nl/rubensten/texifyidea/lang/RequiredArgument.java
package nl.rubensten.texifyidea.lang;

/**
 * @author Sten Wessel
 */
public class RequiredArgument extends Argument {

    RequiredArgument(String name) {
        this(name, Type.NORMAL);
    }

    RequiredArgument(String name, Type type) {
        super(name, type);
    }

    @Override
    public String toString() {
        return "{" + getName() + "}";
    }
}

File: src/nl/hannahsten/texifyidea/lang/commands/RequiredArgument.kt
package nl.hannahsten.texifyidea.lang.commands

/**
 * @author Sten Wessel
 */
@Suppress("ConvertSecondaryConstructorToPrimary")
open class RequiredArgument : Argument {

    @JvmOverloads
    internal constructor(name: String, type: Type = Type.NORMAL) : super(name, type)

    override fun toString() = "{$name}"

    override fun equals(other: Any?) = (other as? RequiredArgument)?.name == this.name

    override fun hashCode() = this.name.hashCode()
}